src/app/index.ts

```diff
@@ -23,10 +23,12 @@ const yo = require("yeoman-generator"); // eslint-disable-line @typescript-eslin
 const childProcessExec = promisify(childProcess.exec);
 const excelCustomFunctions = `excel-functions`;
 let isSsoProject = false;
+let isJsonProject = false;
```

Comment:
I don't like the pattern I'm seeing where there is a new is*Project and project name variable being created for each special case.  This is not a good pattern or precedent that we should continue using.

Reply:
I will try to use `manifestType` then. I think I already started but I decided to continue with the pattern I was seeing on the file.

Reply:
I'm not sure why the manifestType is needed as well.  generator-office should have to know details of the contents of a template.

Reply:
I need some variables in order to reduce the number of questions asked. CFs has specific ones to say to only use excel for example.
I need someway to make it so we don't need questions for Host (only outlook) and for ts vs js (only typescript).

Reply:
Use the size of the supportedHosts array to determine if that question needs to be asked.

This is the first time we've had a case where we aren't saying ts v. js.  We could make a branch off of the js version of the template repo and keep the question, or we could add that to the properties json and use that (in a similar way to the supportedHosts).  Seems like the code should be generic and the differences should be in the properties json.

Reply:
I will change it so we still only have one template (the typescript one) as it will be easier to iterate. We can quickly change if needed later.
I will add that in the next commit

Reply:
We should check for property array sizes and set the value automatically if they are just one instead of making separate variable to track each think.  So something like (this is not exact) "if (jsonData[answerForProjectType.projectType][templates].size == 1) answerForScriptType.scriptType = jsonData[answerForProjectType.projectType][templates][0];" can be used to set the value without asking the question.